### ISSUE TITLE
two column pdf wrong order

### ISSUE THREAD

**Issue by jingangdidi** [jrmuizel/pdf-extract#99]
Thank you for creating this wonderful tool!
When utilizing `extract_text` to extract content from a **two-column PDF** file, the order is incorrectly arranged, with the content on the right column of the first page is followed by that on the right column of the second page). But some two-column PDF file in [arxiv](https://arxiv.org/) no such problem.

**The extracted result:**
![1](https://github.com/user-attachments/assets/55acba5e-295d-49da-9d6c-e2a90dddb67c)

**The raw pdf content:**
![2](https://github.com/user-attachments/assets/74d6ded3-5862-4c6b-9c25-cd0f6a48aa7c)

pdf file: [2_column.pdf](https://github.com/user-attachments/files/18221069/2_column.pdf)

**Comment by jrmuizel** [comment]
Yeah, the library makes no effort to properly assign text to columns yet. The approach I'm thinking of for this would be to gather all the text and positions and then assign columns in a second pass.
